src/plugins/vis_default_editor/public/components/sidebar/sidebar.tsx

```diff
@@ -103,19 +113,47 @@ function DefaultEditorSideBar({
       return;
     }
 
+    const getAggsJSON = (aggs: IAggConfigs) => {
+      return aggs.aggs.map((agg) => {
+        if (agg.type.name === 'filters') {
+          // For filters aggs we need to store each filter to the query log
+          // so that it is made available in the autocomplete.
+          agg.params.filters.forEach((filter: FilterValue) => {
+            const persistedLog = getQueryLog(
+              services.uiSettings,
+              new Storage(window.localStorage),
+              'vis_default_editor',
+              filter.input.language
+            );
+            persistedLog.add(filter.input.query);
+          });
+        }
+        return agg.serialize();
+      });
+    };
```

Comment:
Open to suggestions on a better place to locate this logic, since it feels a little weird to conditionally check the agg type here. Basically what it does is update `localStorage` with the query so that it can be provided by the autocomplete in subsequent inputs.

Previously this was happening inside the filters agg type itself, when writing out the DSL. IMO this doesn't make sense inside aggs as it is purely a UI concern (not to mention the fact that it specifically uses `vis_default_editor` as the appName, and relies on `window.localStorage` which we cannot move to the server).

I thought it would make the most sense putting this inside of the editor's `filters` control, however AFAICT that component doesn't know anything about when changes are applied, which is when this operation needs to take place.

Reply:
as long as its in the UI its fine with me :) @timroes what do you think ?

Reply:
The other idea I had was adding this to the `data.query` service contract, since the `QueryStringInput` component which lives in `data` is the only place that reads out of the persisted log for the autocomplete.

Then usage here would instead be something like:

```ts
    const { services } = useKibana();

    const getAggsJSON = (aggs: IAggConfigs) => {
      return aggs.aggs.map((agg) => {
        if (agg.type.name === 'filters') {
          services.data.query.addFiltersToQueryLog('vis_default_editor`, agg.params.filters);
        }
        return agg.serialize();
      });
    };
```

Then internally the query service can handle the details of instantiating the persisted log, and this keeps control over _where_ the log is stored (in this case localStorage) inside of the data plugin. It also makes it easier for editor & means `getQueryLog` can stay internal to the data plugin

...but either way you're still conditionally performing some side effect for `filters` aggs.

Reply:
I am per se fine with this, if you think it needs to be taken place only when the visualization is applied. But I am curious, why this is the case :-) Meaning if a user types a thing into the first filter of a filters aggregation, they won't get autocomplete for that value when starting the 2nd one? If they want similar filters, wouldn't that be potentially a benefit for them (if we store it e.g. `onBlur`)?

cc @sulemanof if he has additional comments

Reply:
> If they want similar filters, wouldn't that be potentially a benefit for them (if we store it e.g. `onBlur`)?

That's a good point, I hadn't considered `onBlur` but IMO this could work too, and could mean we keep this inside the filters control where it makes more sense anyway.

I'm guessing there was no real reason for doing this during "Apply changes" in the first place, and that it is simply this way because that's when we `write` out the params. 

@sulemanof interested to hear your thoughts on the `onBlur` idea. In the meantime, I'm going to go ahead and create that `addFiltersToQueryLog` function I mentioned, as it feels like it will make things simpler and will keep ownership of writing the items to the persisted log as close to where they are read as possible.

Reply:
I pushed those updates, so the usage looks like this now: 
```ts
    const getAggsJSON = (aggs: IAggConfigs) => {
      return aggs.aggs.map((agg) => {
        if (agg.type.name === 'filters') {
          // For filters aggs we need to store each filter to the query log
          // so that it is made available in the autocomplete.
          agg.params.filters.forEach((filter: FilterValue) => {
            services.data.query.addFilterToQueryLog('vis_default_editor', filter);
          });
        }
        return agg.serialize();
      });
    };
```

Thinking about this more, another option (which would be outside the scope of this PR as it would require changes in a bunch of places), would be to move `<QueryStringInput />` to the plugin contract instead of exporting statically, and then handle all of this internally `onBlur`, adding props to enable everything, e.g.
```js
start(core, { data }) {
  const { QueryStringInput } = data.ui;
  render(<QueryStringInput appName="vis_default_editor" addValueToAutocomplete={true} />);
  // then input values would be automatically added to the query log `onBlur`
}
```

Reply:
I fully support @timroes suggestion on moving this into filters control itself and save it on blur event.
It will remain the sidebar clear of such a special case.
I also want to add here that both the app name and the localStorage lives in kibana context, so you could use it from there.

Wraps the whole editor:

![image](https://user-images.githubusercontent.com/31325372/86343457-a851bc80-bc61-11ea-9ce7-ad781c67b928.png)

Reply:
Updated. Everything is handled in the filters control now `onBlur`, using that helper I exposed from data.

Long term I'd like to explore fully containing this inside `<QueryStringInput />`, but for now this feels like a clean enough separation of concerns.